File: src/network/lobby/packets/user/LoginUserPacket.java
package network.lobby.packets.user;

import misc.compress.Compressor;
import network.lobby.packets.UserPacket;

public class LoginUserPacket extends UserPacket
{
	private String name;
	private int rank;

	public LoginUserPacket(String name, int rank)
	{
		this.name = name;
		this.rank = rank;
	}

	public void assign(LoginUserPacket loginPacket)
	{
		this.name = loginPacket.name;
		this.rank = loginPacket.rank;
	}

	public LoginUserPacket(LoginUserPacket packet)
	{
		name = new String(packet.getName());
		rank = packet.getRank();
	}

	@Override public byte getCID() { return Compressor.LOGIN_USER_PACKET_CID; }
	@Override public byte[] compress()
	{
		return Compressor.concat(new byte[][]{
			Compressor.compressString(getName()),
			Compressor.compressInt(getRank())});
	}

	public String getName() { return name; }
	public int getRank() { return rank; }
}

File: cpp/src/network/packets/LoginUserPacket.cpp
#include "LoginUserPacket.hpp"

#include <misc/compress/CompressBuffer.hpp>

LoginUserPacket::LoginUserPacket(const std::string& name, int rank) : name(name), rank(rank)
{}

LoginUserPacket::LoginUserPacket(CompressBuffer* buffer)
	: name(buffer->cutString()), rank(buffer->cutInt())
{}

std::string LoginUserPacket::getCompressString() const
{
	return compressString(getName()) + compressInt(getRank());
}

CID LoginUserPacket::getCID() const
{
	return LOGIN_USER_PACKET_CID;
}

std::string LoginUserPacket::getName() const
{
	return name;
}

int LoginUserPacket::getRank() const
{
	return rank;
}
